beacon_node/eth2_libp2p/src/rpc/mod.rs

```diff
@@ -59,7 +60,7 @@ pub enum RPCReceived<T: EthSpec> {
     ///
     /// The `SubstreamId` is given by the `RPCHandler` as it identifies this request with the
     /// *inbound* substream over which it is managed.
-    Request(SubstreamId, RPCRequest<T>),
+    Request(SubstreamId, RpcRequestContainer<T>),
```

Comment:
Yeah realised now that it's not required. I had implemented the `expected_responses`, `protocol` and `stream_termination` methods on the `RpcRequestContainer` instead of `RPCRequest`. Because of that, I ended up using the container in places where it was not required. 
Thanks for pointing it out. Fixed now :)